polyfactory/factories/base.py

```diff
@@ -774,6 +780,20 @@ def should_set_none_value(cls, field_meta: FieldMeta) -> bool:
             and create_random_boolean(random=cls.__random__)
         )
 
+    @classmethod
+    def should_use_default_value(cls, field_meta: FieldMeta) -> bool:
```

Comment:
[Nit] Now the call live at the same `should_set_field_value` could these be combined?

Reply:
I did consider that, but that's supposed to be more for like checking if it's a private variable or not. At least, that's what the actual usage is. My worry is that users may override the behavior and then not include the check for whether the default should be used or not and then end up with unexpected behavior.